### ISSUE TITLE
communication between devtools and content script

### ISSUE THREAD

**Issue by HuiwenHong** [binaryage/chromex#8]
I've tried to connect background and devtools following the chromex-sample and it worked. But when I tried the same way to pass messages from a devtools extension to content script in the inspected tab, I'm getting the error shown below. 

<img width="762" alt="screen shot 2017-11-09 at 11 45 27 am" src="https://user-images.githubusercontent.com/11634072/32617662-900a88b8-c543-11e7-8826-c347f7280eba.png">

Is it the correct way to set up the messaging between devtools and content script? or is there another way? I put my code below. 

**devtool**
```clojure
(defn connect-to-content-script! [tab-id]
  (let [content-script-port (tabs/connect tab-id)]
    (log "content-script-port" content-script-port)
    (post-message! content-script-port "hello from DEVTOOLS!")))

;; -- main entry point 
(defn init! []
  (log "DEVTOOL-PANEL: init!")
  (log "DEVTOOL-PANEL: tab id " (get-tab-id))
  (connect-to-content-script! (get-tab-id)))
```

**content script**

```clojure
(defn run-chrome-event-loop! [chrome-event-channel]
  (log "CONTENT SCRIPT: starting main event loop...")
  (go-loop [event-num 1]
           (when-some [event (<! chrome-event-channel)]
             (log "CONTENT SCRIPT: get event " event-num " | " event)
             (recur (inc event-num)))
           (log "CONTENT SCRIPT: leaving main event loop")))

(defn boot-chrome-event-loop! []
  (let [chrome-event-channel (make-chrome-event-channel (chan))]
    (runtime/tap-all-events chrome-event-channel)
    (run-chrome-event-loop! chrome-event-channel)))

;; -- main entry point 
(defn init! []
  (log "CONTENT SCRIPT: init")
  (boot-chrome-event-loop!))
```

**Comment by darwin** [comment]
From the code above it looks that you copy&pasted code from background page into your content script and expected it to work. This is misunderstanding [how content scripts work](https://developer.chrome.com/extensions/content_scripts). They don't have access to chrome APIs, they run code in the context of a page.

Proper content script code should look like this:
https://github.com/binaryage/chromex-sample#content-script

The error is probably raised on line `(runtime/tap-all-events chrome-event-channel)` which tries to access `chrome.runtime.onStartup` from your content script (page context).

**Comment by HuiwenHong** [comment]
Thanks for the quick response! 
 
I checked the link you mentioned on content script and it says content script has access to some of the `chrome.runtime` API but not `onStartup`. I found I can use `runtime/tap-on-connect-events` instead of `runtime/tap-all-events` and it works perfectly. So I'm able to connect devtools and content script directly now.

I've been enjoying your library to create chrome extensions, great job!
